### PR TITLE
Refactor to accept & mount multiple EFSs

### PR DESCRIPTION
**Pull request by jeffbyrnes**
Preserves backwards-compatibilty by early-outing to using a single `node['ds_efs']['id']` if provided. It retains the `node['ds_efs']['path']` attribute, but _only_ if you also use the older style `node['ds_efs']['id']`. I.e., if you’re using the old behavior, it’s in use consistently.

Dropping that attribute, and providing one or more pairs via a new attribute in the form of a hash, will trigger the new behavior:

```
node['ds_efs']['filesystems'] = {
  'arbitrary_name' => {
    'id' => 'fs-one',
    'path' => '/mnt/efs/one',
  },
  'another_arbitrary_name' => {
    'id' => 'fs-two',
    'path' => '/mnt/efs/two',
  },
}
```

[edited for clarity, per the discussion below]

### CODE REVIEW

**Comment by jeffbyrnes** [comment]
Thanks! I’ve discovered an issue with this, so I’m still working it out. Stand by for additional review requests.

File: recipes/default.rb [hunk 1]
```diff
@@ -21,21 +29,30 @@
...
+  mount path do
+    device  "#{id}:/"
+    fstype  'efs'
+    options 'defaults,_netdev'
+    dump    0
+    pass    0
+    action  %i(mount enable)
+  end
+
+  include_recipe 'ds_common_basics::user'
```
Comment (nonrenewable) [inline recipes/default.rb]: do we want this to execute for every EFS or just once overall? (i haven't looked into it, figured you'd just know the answer 🙂 )

Comment (jeffbyrnes) [reply recipes/default.rb]: Good catch. Just once! In fact, I’d love to come up with a way to do this with a non-propriety recipe; this is a nice open-source cookbook. Next refactor, I think.

**Comment by jeffbyrnes** [comment]
@nonrenewable @jareyes @ironwallaby appreciate the review of the code. Would one/some/all of you give some feedback on the questions I posit in the description?

I want to get this right from the jump, in case we need it, and swapping that hash around would definitely be painful, owing to the rather far-reaching nature of this cookbook.

**Comment by jeffbyrnes** [comment]
Ok, thanks team. Per some AFK convo, I’m going to tweak this so our hash looks like this:

```ruby
node['ds_efs']['filesystems'] = {
  'key' => {
    'id' => 'fs-id',
    'path' => '/mnt/efs/mountpoint',
}
```

That way, it’s easy to reason about where your EFS will end up, and easy to pass things in, so if ever you need to get the filesystem ID of an EFS, you can do this: `node['ds_efs']['filesystems']['stations']['path']`, which is similar to how we can currently ask for `node['ds_efs']['path']`.

**Pull request merged**